material_3_demo/web/index.html

```diff
@@ -38,22 +38,29 @@
   </script>
   <!-- This script adds the flutter initialization JS code -->
   <script src="flutter.js" defer></script>
+  <script type="text/javascript" src="assets/packages/web_startup_analyzer/lib/web_startup_analyzer.js"></script>
 </head>
 <body>
-  <script>
-    window.addEventListener('load', function(ev) {
-      // Download main.dart.js
-      _flutter.loader.loadEntrypoint({
-        serviceWorker: {
-          serviceWorkerVersion: serviceWorkerVersion,
-        },
-        onEntrypointLoaded: function(engineInitializer) {
-          engineInitializer.initializeEngine().then(function(appRunner) {
-            appRunner.runApp();
-          });
-        }
-      });
+<script>
+  var analyzer =  new FlutterWebStartupAnalyzer();
+
+  window.addEventListener('load', function(ev) {
+    analyzer.markStart("loadEntrypoint");
+    _flutter.loader.loadEntrypoint({
+      serviceWorker: {
+        serviceWorkerVersion: serviceWorkerVersion,
+      },
+      onEntrypointLoaded: function(engineInitializer) {
+        analyzer.markFinished("loadEntrypoint");
+        analyzer.markStart("initializeEngine");
+        engineInitializer.initializeEngine().then(function(appRunner) {
+          analyzer.markFinished("initializeEngine");
+          analyzer.markStart("appRunnerRunApp");
```

Comment:
Is `"appRunnerRunApp"` special? Is it wired into the framework?

Reply:
@johnpryan – thoughts here?

Reply:
Sorry, I missed this comment. These labels are arbitrary - they're only used to call into the web's performance API to mark timing information into the Performance timeline.

Reply:
is `appRunnerRunApp` ever marked finished? I didn't see it

Reply:
It's marked as finished on the Dart side, when the`WebStartupAnalyzer` class is constructed in `main()`.

Reply:
There could be a smarter way to do this, this is just what I hacked together

Reply:
cc: @ditman